### ISSUE TITLE
Find/write alternative to japplemenubar and remove it

### ISSUE THREAD

**Issue by benfry** [benfry/processing4#221]
In the [macOS build](https://github.com/processing/processing4/tree/master/build/macosx) we have a dependency on [japplemenubar](https://github.com/kritzikratzi/jAppleMenuBar) which hasn't been updated in a decade. I'm concerned that it will stop working at any time, because it uses `SetSystemUIMode`, which is a Carbon API and we're supposed to be using `setPresentationOptions()` (a Cocoa API) instead. In fact, I'm not even sure why it still works.

There's the code in question: https://github.com/kritzikratzi/jAppleMenuBar/blob/master/src/native/jAppleMenuBar.m

The update to this code was done in Mozilla: https://bugzilla.mozilla.org/attachment.cgi?id=8616917&action=diff in which this code:

```objective-c
if (sMenuBarHiddenCount > 0) {
  ::SetSystemUIMode(kUIModeAllHidden, 0);
} else if (sDockHiddenCount > 0) {
  ::SetSystemUIMode(kUIModeContentHidden, 0);
} else {
  ::SetSystemUIMode(kUIModeNormal, 0);
}
```

was changed to 

```objective-c
NSApplicationPresentationOptions options;
if (sMenuBarHiddenCount > 0) {
  options = NSApplicationPresentationHideDock | NSApplicationPresentationHideMenuBar;
} else if (sDockHiddenCount > 0) {
  options = NSApplicationPresentationHideDock;
} else {
  options = NSApplicationPresentationDefault;
}
[NSApp setPresentationOptions:options];
```

In our case, we don't need any of this logic, we need:

```objective-c
NSApplicationPresentationOptions options = 
  NSApplicationPresentationHideDock | NSApplicationPresentationHideMenuBar;
[NSApp setPresentationOptions:options];
```

and a second call to clear those out.

There's also [this version](https://developer.apple.com/forums/thread/124849) done in Swift that toggles, though we'd be changing `autoHideMenuBar` to just `hideMenubar` and `autoHideDock` to `hideDock`. 

```swift
let application = NSApplication.shared
let autoHidden = application.presentationOptions.contains(.autoHideMenuBar)
if autoHidden {
    application.presentationOptions = []
} else {
    application.presentationOptions = [.autoHideDock, .autoHideMenuBar]
}
```

Reference for [presentationOptions](https://developer.apple.com/documentation/appkit/nsapplication/1428664-presentationoptions) and [hideMenuBar](https://developer.apple.com/documentation/appkit/nsapplication/presentationoptions/1428487-hidemenubar). 

Anyhow, we need something as simple as the japplemenubar solution (a teeny tiny JNI library) or something that's like two lines of JNA that handles this for us. In particular, we'd want to avoid adding an entire dependency like Rococoa or similar.

A JNA solution would mean adding JNA to core, but we may need to do that soon in order to get native screen resolutions for handling hidpi screens.

Or if not, it'd be great to stay away from JNA, maybe we do the JNI one liner. Or maybe the hidpi calls can mostly be handled with similar one-liner JNA code, which we could use to build out a tiny native sidecar for core.

**Comment by knupel** [comment]
I don't if that's can help or poluate the topic, because my understanding of what is necessary to help. But just in case I work around an  apple menu bar for my library rope. https://github.com/StanLepunK/Rope/blob/master/src/rope/gui/bar/R_Apple_Bar.java I've used javax.swing.JFrame, jMenu, JMenuBar et cetera. If it's a possible option maybe I can try to understand deeply what is necessary for Processing 4 :)

**Comment by benfry** [comment]
Thanks; it looks like that's solving a different type of menu-related issue, but I appreciate the interest.

**Comment by knupel** [comment]
Too bad, maybe next time I can add a stone to the Processing castle !

**Comment by benfry** [comment]
:)

**Comment by benfry** [comment]
Resolved for 4.0 beta 5.

**Comment by benfry** [comment]
Purged the old code for beta 7, now with a cleaned up version of the rewrite started in beta 5: https://github.com/processing/processing4/commit/4ab1c8cda57263294f04cb6c72cb849f01d77df0

Documentation here: https://github.com/processing/processing4/tree/master/core/different
